lib/webmachine/request.rb

```diff
@@ -164,6 +165,12 @@ def options?
       method == OPTIONS_METHOD
     end
 
+    ROUTING_PATH_MATCH = /^\/(.*)/.freeze
+
+    def routing_tokens
```

Comment:
I'm a fan of making objects as immutable as possible. What do you think about making routing_tokens an optional argument to the Request class, and when it is not passed in, the constructor sets it to be `uri.path.match(ROUTING_PATH_MATCH)[1].split(SLASH)`? That way we can take the writer off the routing_tokens attribute.